woocommerce/woocommerce
# Fix pnpm version to 7.11.0 in engines and in CI

### All Submissions:






### Changes proposed in this Pull Request:

The problem:

Some people have been seeing issues with pnpm 7.12.x, and also some have not realized they should upgrade to pnpm 7 from 6. To solve this I've reintroduced engines but I've also consolidated engines across all relevant packages so no one will accidentally use the wrong version of pnpm without being told so!

I've also fixed the pnpm version in CI to ensure consistency.

Note: I'm still not sure about the stability of 7.12.x. But once I feel more comfortable we could loosen the restriction of this version to 7.x or similar.

### How to test the changes in this Pull Request:

1. try install a non supported version of pnpm: `npm i -g pnpm@7.12.0`
2. try `pnpm i` at root. It should fail and tell you you're using the wrong version
3. Now switch back to correct version `npm i -g pnpm@7.11.0` and run `pnpm i` it should succeed
4. Also a `pnpm run build` should still succeed with correct pnpm version.



### FOR PR REVIEWER ONLY:
